A66-otel-stats.md

```diff
@@ -0,0 +1,305 @@
+# OpenTelemetry Metrics
+
+*   Author: Yash Tibrewal (@yashykt)
+*   Approver: Mark Roth (@markdroth)
+*   Status: In Review
+*   Implemented in: <language, ...>
+*   Last updated: Jul 20, 2023
+*   Discussion at: https://groups.google.com/g/grpc-io/c/po-deqYEQzE
+
+## Abstract
+
+Propose a metrics data model for gRPC OpenTelemetry metrics.
+
+## Background
+
+There are a collection of
+[metrics](https://github.com/census-instrumentation/opencensus-specs/blob/master/stats/gRPC.md)
+proposed by OpenCensus for gRPC. OpenCensus is no longer being actively
+maintained and is being
+[deprecated](https://opentelemetry.io/blog/2023/sunsetting-opencensus/#:~:text=Compatibility%20specification%204.-,What%20to%20Expect%20After%20July%2031st%2C%202023,found%20will%20not%20be%20patched.),
+with OpenTelemetry suggested as the successor framework.
+
+### Related Proposals:
+
+*   [gRPC Retry Stats](A45-retry-stats.md)
+
+## Proposal
+
+### Units
+
+Following the
+[OpenTelemetry Metrics Semantic Conventions](https://opentelemetry.io/docs/specs/otel/metrics/semantic_conventions/),
+the following units are used -
+
+*   Latencies are measured in float64 seconds, `s`
+*   Sizes are measured in bytes, `By`
+*   Counts for number of calls are measured in `{call}`
+*   Counts for number of attempts are measured in `{attempt}`
+
+Buckets for histograms in default views should be as follows -
+
+*   Latency : 0, 0.00001, 0.00005, 0.0001, 0.0003, 0.0006, 0.0008, 0.001, 0.002,
+    0.003, 0.004, 0.005, 0.006, 0.008, 0.01, 0.013, 0.016, 0.02, 0.025, 0.03,
+    0.04, 0.05, 0.065, 0.08, 0.1, 0.13, 0.16, 0.2, 0.25, 0.3, 0.4, 0.5, 0.65,
+    0.8, 1, 2, 5, 10, 20, 50, 100
+*   Size : 0, 1024, 2048, 4096, 16384, 65536, 262144, 1048576, 4194304,
+    16777216, 67108864, 268435456, 1073741824, 4294967296
+*   Count : 0, 1, 2, 4, 8, 16, 32, 64, 128, 256, 512, 1024, 2048, 4096, 8192,
+    16384, 32768, 65536
+
+### Attributes
+
+*   `grpc.method` : Full gRPC method name, including package, service and
+    method, e.g. "google.bigtable.v2.Bigtable/CheckAndMutateRow"
+*   `grpc.status` : gRPC server status code received, e.g. "OK", "CANCELLED",
+    "DEADLINE_EXCEEDED"
+*   `grpc.target` : Target URI used when creating gRPC Channel, e.g.
+    "dns:///pubsub.googleapis.com:443", "xds:///helloworld-gke:8000"
+*   `grpc.authority` : Authority used by the call/attempt, e.g.
+    "pubsub.googleapis.com", "helloworld-gke"
```

Comment:
In most cases the `grpc.authority` is going to be the same as the authority part of `grpc.target` (unless override host is used on a call) so there is another redundancy here. 

Also if your examples are supposed to match the previous example for `grpc.target` wouldn't the correct values be `pubsub.googleapis.com:443` and `helloworld-gke:8000` i.e. include the port?

Reply:
Removed `authority`. Explanation below.